src/object.d

```diff
@@ -285,8 +285,7 @@ class TypeInfo
 
     /// Return default initializer.  If the type should be initialized to all zeros,
     /// an array with a null ptr and a length equal to the type size will be returned.
-    // TODO: make this a property, but may need to be renamed to diambiguate with T.init...
-    const(void)[] init() nothrow pure const @safe @nogc { return null; }
+    abstract const(void)[] init() nothrow pure const @nogc;
```

Comment:
Just tested it out, it actually does the right thing -- derived functions are inferred `@safe` unless marked with `@trusted`, so go ahead and mark `@safe` and then you can mark the derived functions that need it with `@trusted`